### ISSUE TITLE
dedupe: false

### ISSUE THREAD

**Issue by Akryum** [nuxt/nuxt#30370]
### Describe the feature

Make it possible to disable deduplication of the requests. Useful for a throttled search for example.

```ts
const search = ref('')

const throttledSearch = useThrottle(search, 500)

const { data } = useFetch('/api/search', {
  query: computed(() => ({ text: throttledSearch.value })),
  immediate: !!throttledSearch.value,
  dedupe: false, // Here
})
```

All requests would be made and update `data` to update the search results as the user types.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [x] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [x] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by cernymatej** [comment]
We'd have to keep track of the promises and ensure that we don't end up with stale data if an older request took longer than a newer one.

Personally, when I’m searching for something, I prefer to see the most recent results rather than older ones. What’s the issue with the default `'cancel'` behavior? If a request finishes before the next one, we can still achieve the desired behavior of displaying search results as the user types.

**Comment by Akryum** [comment]
The issue is that if the throttle is 500ms and the requests all take 600ms, the results will never get updated until the last request is done.

**Comment by cernymatej** [comment]
Yes, I get that, but is it really a real world use case? Provided we don't have poor internet connection, what request takes that long?

As a user, I would expect a loading indicator to be shown while the search requests are pending. Now let's say that there are some results visible, I keep typing and another request (let's call it B) is made. Then, while the request B is in progress, I finish typing. A third request (C) is made - this is the final search query. 200ms after I finished typing, request B finishes and the search results are replaced with the "almost" complete results and the loading indicator is still shown. Only after another while does the request C with the final search results finish, the search results are replaced once again and the loading indicator is hidden.
Personally, I'm not a fan of the content flashing. Maybe there is another way we could solve this?
Just my thoughts on this. I'm interested in what you think.

**Comment by Akryum** [comment]
> what request takes that long?

The GitHub API for example :D
